include/nbl/ext/TextRendering/TextRendering.h

```diff
@@ -0,0 +1,232 @@
+	uint16_t textureAtlasGlyphIndex;
+};
+
+struct StringBoundingBox
+{
+	SPixelCoord min, max;
+};
+
+class NBL_API TextRenderer
+{
+public:
+	typedef typename uint32_t size_type;
+
+	TextRenderer(core::smart_refctd_ptr<ILogicalDevice>&& device, uint32_t maxGlyphCount, uint32_t maxStringCount, uint32_t maxGlyphsPerString);
+
+	using pool_size_t = uint32_t;
+
+	using glyph_geometry_pool_t = video::SubAllocatedDataBufferST<pool_size_t>;
+	// Data stored as SoA in property pool:
+	// - Glyph offset
+	// - String bounding box
+	using string_pool_t = video::CPropertyPool<core::allocator, pool_size_t, StringBoundingBox>;
+
+	struct string_handle_t
+	{
+		pool_size_t stringAddr = core::address_type_traits<pool_size_t>::invalid_address;
+		pool_size_t glyphDataAddr = core::address_type_traits<pool_size_t>::invalid_address;
+		pool_size_t glyphCount = 0u;
+	};
+
+	template<class Clock = typename std::chrono::steady_clock>
+	uint32_t allocateStrings(
+		const std::chrono::time_point<Clock>& maxWaitPoint, // lets have also a version without this (default)
+		const uint32_t count, // how many strings
+		string_handle_t* handles, // output handles, if `glyphDataAddr` was not primed with invalid_address, allocation will not happen, likewise for `stringDataAddr`
+		const char* const* stringData,
+		const StringBoundingBox* wrappingBoxes = nullptr // optional, to wrap paragraphs
+	);
+
+	inline uint32_t allocateStrings(
+		const uint32_t count, // how many strings
+		string_handle_t* handles, // output handles, if `glyphDataAddr` was not primed with invalid_address, allocation will not happen, likewise for `stringDataAddr`
+		const char* const* stringData,
+		const StringBoundingBox* wrappingBoxes = nullptr // optional, to wrap paragraphs
+	)
+	{
+		return allocateStrings(GPUEventWrapper::default_wait(), count, handles, stringData, wrappingBoxes);
+	}
+
+	void freeStrings(
+		const uint32_t count, // how many strings
+		const string_handle_t* handles,
+		core::smart_refctd_ptr<IGPUFence>&& fence // for a deferred free
+	);
+
+	// One of the sets would be our global DS, and the other would be the user provided visible strings DS
+	core::smart_refctd_ptr<video::IGPUGraphicsPipeline> createPipeline(
+		video::IGPUObjectFromAssetConverter::SParams& cpu2gpuParams, 
+		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> visibleStringLayout
+	);
+
+	core::smart_refctd_ptr<video::IGPUGraphicsPipeline> createPipelineIndexed(
+		video::IGPUObjectFromAssetConverter::SParams& cpu2gpuParams, 
+		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> visibleStringLayout
+	);
+
+	// Visible strings are provided by the user's culling system
+	void updateVisibleStringDS(
+		core::smart_refctd_ptr<video::IGPUDescriptorSet> visibleStringDS,
+		core::smart_refctd_ptr<video::IGPUBuffer> visibleStrings,
+		core::smart_refctd_ptr<video::IGPUBuffer> visibleStringGlyphCounts,
+		core::smart_refctd_ptr<video::IGPUBuffer> cumulativeGlyphCount
+	);
+
+	void prefixSumHelper(
```

Comment:
@devshgraphicsprogramming We don't need this anymore since user is appending + prefixSumming at the same time?
Or the signature should change to something like `appendAndPrefixSumHelper(...)`